File: LeetCode/467环绕字符串中唯一的子字符串.rb
# @param {String} p
# @return {Integer}
def find_substring_in_wrapround_string(p)
    temp = 0
    arr = Array.new(26, 0)
    (0...p.size).each do |x|
        if x > 0 && (p[x].ord - p[x - 1].ord + 26) % 26 == 1 then
            temp += 1
        else 
            temp = 1
        end
        arr[p[x].ord - 97] = [arr[p[x].ord - 97], temp].max
    end
    return arr.sum
end

File: LeetCode/467环绕字符串中唯一的子字符串.cpp
class Solution {
public:
    int findSubstringInWraproundString(string p) {
        vector<int> dp(26);
        int temp = 0;
        for (int i = 0; i < p.size(); i++) {
            if (i && (p[i] - p[i - 1] + 26) % 26 == 1) temp++;
            else temp = 1;
            dp[p[i] - 'a'] = max(dp[p[i] - 'a'], temp);
        }
        return accumulate(dp.begin(), dp.end(), 0);
    }
};